src/population/visualization/family_tree_tracker.py

```diff
@@ -0,0 +1,201 @@
+from datetime import datetime
+import json
+import networkx as nx
+import os
+import numpy as np
+from loguru import logger
+from population.visualization.visualization import create_and_save_figure, make_dir_if_not_exists
+from population.visualization.graphing_model import get_neural_net_positions
+from population.visualization.gene_data_processing import convert_genes_to_numerical, get_pca_positions, get_pca_colors
+
+class FamilyTreeTracker:
+    """
+    A class used to track relations between nodes, along with other attributes.
+    This saves the genome data as a series of lines in a temporary file, and
+    """
+
+    def __init__(self, temp_file_dir: str='temp_genome_data', delete_temp_file: bool=True):
+
+        # whether you delete it at the end
+        self._delete_temp_file = delete_temp_file
+
+        # create a temporary filename to store genome data in
+        current_time = datetime.now().strftime("%Y-%m-%d_%H-%M-%S" + str(np.random.randint(0, 1)))
+
+        # keeps a string reference to the directory that temporary files are stored in
+        self.temp_file_dir = temp_file_dir
+
+        # stores a string  reference to the filepath of the temporary file
+        self.family_tracker_file = os.path.join(
+            temp_file_dir,
+            f'temp_genome_data_{current_time}_{str(np.random.randint(1_000, 9_999))}')
+
+        # make the temporary file directory if it doesn't already exist
+        make_dir_if_not_exists(temp_file_dir)
+
+    def track_genomes(self, genomes: list):
+        """
+        Track a list of genomes by adding them to a temporary file.
+
+        Args:
+            genomes (list): A list of genomes to track.
+        """
+        # make sure it is actually a list
+        if (genomes is not None) and hasattr(genomes, '__len__'):
+            with open(self.family_tracker_file, 'a') as f:
+
+                # iterate over all genomes
+                for genome in genomes:
+
+                    # store the json data as another line that is appended to the end
+                    f.write(json.dumps(genome.to_dict()) + '\n')
+        else:
+            raise TypeError(f"Object of type {type(genomes)} does not support __len__()")
+
+    def load_genomes(self):
+        """
+        Load all genomes as a graph.
+
+        Returns:
+            graph (nx.DiGraph): A graph of genomes and their family relations.
+            node_genes (dict[int, list[int]]): The dict of node genes for every genome.
+            edge_genes (dict[int, list[int]]): The dict of edge genes for every genome.
+            fitnesses (dict[int, float]): This dict has a fitness for every genome.
+        """
+
+        # never hurts to be sure it actually exists
+        if os.path.exists(self.family_tracker_file):
```

Comment:
This actually makes the return type optional since there is no return if the path doesn't exist.